### PR TITLE
Adding typings for moment-holiday

### PR DESCRIPTION
**Pull request by rwdalpe**
PR upstream to add typings in the library has been open
for a while <https://github.com/kodie/moment-holiday/pull/14>
without feedback.

Adding the typings here until that PR changes state.

Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [X] The package does not provide its own types, and you can not add them.
- [X] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [X] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [X] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Review by plantain-00 — APPROVED** [review]

**Comment by RyanCavanaugh** [comment]
Approved by third party. PR appears ready to merge pending review by a maintainer.

File: types/moment-holiday/moment-holiday-tests.ts
Comment (DanielRosenwasser) [inline types/moment-holiday/moment-holiday-tests.ts]: Can you use the `import moment = require('moment')` syntax, and explicitly import `moment-holiday` as well?

Comment (rwdalpe) [reply types/moment-holiday/moment-holiday-tests.ts]: Sure thing!

**Comment by DanielRosenwasser** [comment]
Thanks!

**Comment by rwdalpe** [comment]
Thank you!

**Pull request merged**
